librepo/downloader.c

```diff
@@ -1464,6 +1464,20 @@ prepare_next_transfer(LrDownload *dd, gboolean *candidatefound, GError **err)
     }
     target->curl_handle = h;
 
+    // Append the LRO_ONETIMEFLAG if instructed to do so
+    LrHandle *handle = target->handle;
+    if (handle && handle->onetimeflag && handle->otf_context) {
+        char *sep = "?";
+        if (g_strrstr(full_url, sep) != NULL) sep = "&";
```

Comment:
Can we please not do oneliner `if` conditionals, they're hard to read...

Reply:
Fixed.